dwave/preprocessing/src/presolveimpl.hpp

```diff
@@ -65,7 +66,53 @@ class PresolverImpl {
         return model_.detach_model();
     }
 
-    const Feasibility& feasibility() const { return feasibility_; }
+    const Feasibility& feasibility() const { return model_.feasibility; }
+
+    // Get the maximal activity constributed by variable v
+    static bias_type maximal_activity(const constraint_type& constraint, index_type v) {
+        assert(constraint.is_linear());  // todo: support quadratic
```

Comment:
Seems like this should be an exception which throws a useful message, like "Quadratic support is not implemented yet."

Reply:
testing linearity is O(n) so rather expensive to do here, and because this is not a user-facing function, we use an assert (which will show up in testing but not in prod). For user-facing functions we're more likely to use errors.